lib/readline.js

```diff
@@ -580,27 +591,57 @@ Interface.prototype._wordLeft = function() {
 Interface.prototype._wordRight = function() {
   if (this.cursor < this.line.length) {
     var trailing = this.line.slice(this.cursor);
-    var match = trailing.match(/^(?:\s+|\W+|\w+)\s*/);
+    var match = trailing.match(/^(?:\s+|[^\w\s]+|\w+)\s*/);
     this._moveCursor(match[0].length);
   }
 };
 
+function codePointLeftOf(i, str) {
+  // obtain the code point to the left
+  let previousChar;
+  let len = 0;
+  for (const char of str) {
+    len += char.length;
+    previousChar = char;
+    if (len === i) {
+      return previousChar;
+    }
+  }
+}
```

Comment:
AFAIK this could be rewritten to `O(1)` (all code points above 2 ** 16 should have a length of two, all others one):

```js
function charLengthLeft(str, i) {
  if (i <= 0)
    return 0;
  if (i > 1 && str.codePointAt(i - 2) > 2 ** 16 ||
      str.codePointAt(i - 1) > 2 ** 16) {
    return 2;
  }
  return 1;
}
```

(I also changed the return value which reduces the code on the receiving side and changed the behavior a tiny bit to not fail if `i` is in-between a code point of length 2)

Reply:
It maybe faster due to using a built in function, but I am unaware of any optimizations that allow `O(1)`indexing  of code points . I.E. I believe each `codePointAt` is `O(n)`. Is the JIT seeing an optimization  I am not?

Reply:
@Avi-D-coder I did not check the internals in V8 so you are probably right about that. However, benchmarking this seems has a constant time (or near constant) for different string sizes and `i` for me while the other implementation increases is clearly `O(n)`. That's why I draw that conclusion.

Reply:
It might be possible that the @nodejs/v8 team is able to optimize the loop version further.